src/contract.py

```diff
@@ -39,6 +39,9 @@ def to_dict(self) -> dict[str, any]:
             'observations': self.num_mon
         }
 
+    def set_num_mon(self, num_mon) -> list[float]:       # used by BB Barrier pricer
```

Comment:
I would simply go with setting this property whereever you need. There's no point of having this setter method.

Reply:
I'm not against setting the number of monitoring points to a smaller number for barrier options. It's just useless to do it via a method in the current framework. Just use `contract.num_mon = number`.

Reply:
Indeed, but originally everything was with underscore.